frontends/p4/fromv1.0/converters.cpp

```diff
@@ -716,6 +700,45 @@ class ComputeCallGraph : public Inspector {
     }
 };
 
+/// Table call graph should be built after the control call graph is built.
+/// In the case that the program contains an unused control block, the
+/// table invocation in the unused control block should not be considered.
+class ComputeTableCallGraph : public Inspector {
+    ProgramStructure *structure;
+
+ public:
+    explicit ComputeTableCallGraph(ProgramStructure *structure) : structure(structure) {
+        CHECK_NULL(structure);
+        setName("ComputeCallGraph");
+    }
+
+    void postorder(const IR::Apply *apply) override {
+        LOG3("Scanning " << apply->name);
+        auto tbl = structure->tables.get(apply->name.name);
+        if (tbl == nullptr)
+            ::error("Could not find table %1%", apply->name);
+        auto parent = findContext<IR::V1Control>();
+        BUG_CHECK(parent != nullptr, "%1%: Apply not within a control block?", apply);
+
+        auto ctrl = get(structure->tableMapping, tbl);
+
+        // skip control block that is unused.
+        if (!structure->calledControls.isCallee(parent->name) &&
+            parent->name != P4V1::V1Model::instance.ingress.name &&
+            parent->name != P4V1::V1Model::instance.egress.name )
+            return;
+
+        if (ctrl != nullptr && ctrl != parent) {
+            auto previous = get(structure->tableInvocation, tbl);
+            ::error("Table %1% invoked from two different controls: %2% and %3%",
+                    tbl, apply, previous);
```

Comment:
Should we check for mutually exclusive control paths? I understand that this may be future work.